runtime/particle-spec.js

```diff
@@ -18,9 +18,9 @@ class ConnectionSpec {
     this.rawData = rawData;
     this.direction = rawData.direction;
     this.name = rawData.name;
-    this.typeName = rawData.type;
-    this.typeName = typeLiteral.convertNamedVariablesToVariables(this.typeName, typeVarMap);
-    this.type = new runtime.internals.Type(this.typeName);
+    let type = rawData.type;
+    type = typeLiteral.convertNamedVariablesToVariables(type, typeVarMap);
```

Comment:
we can lift convertNamedVariables to Type as well, then we don't need the type literal library at all